clients/vault/src/oracle/collector/proof_builder.rs

```diff
@@ -231,29 +278,33 @@ impl ScpMessageCollector {
 	fn get_txset_from_horizon_archive(&self, slot: Slot) -> impl Future<Output = ()> {
 		tracing::warn!("Fetching TxSet for slot {} from horizon archive", slot);
 		let txset_map_arc = self.txset_map_clone();
-		let stellar_history_base = self.stellar_history_base_url();
+		let archive_urls = self.stellar_history_archive_urls();
+
 		async move {
-			let tx_archive_storage = TransactionsArchiveStorage(stellar_history_base);
-			let transactions_archive = tx_archive_storage.get_archive(slot).await;
+			for archive_url in archive_urls {
+				let tx_archive_storage = TransactionsArchiveStorage(archive_url);
+				let transactions_archive = tx_archive_storage.get_archive(slot).await;
 
-			if let Err(e) = transactions_archive {
-				tracing::error!(
+				if let Err(e) = transactions_archive {
+					tracing::error!(
 					"Could not get TransactionsArchive for slot {slot:?} from horizon archive: {e:?}"
 				);
-				return
-			}
-			let transactions_archive: XdrArchive<TransactionHistoryEntry> =
-				transactions_archive.unwrap();
-
-			let value = transactions_archive
-				.get_vec()
-				.iter()
-				.find(|&entry| Slot::from(entry.ledger_seq) == slot);
-
-			if let Some(target_history_entry) = value {
-				tracing::debug!("Adding archived tx set for slot {}", slot);
-				let mut tx_set_map = txset_map_arc.write();
-				tx_set_map.insert(slot, target_history_entry.tx_set.clone());
+					return
```

Comment:
Correct me if I'm wrong... 
let's say we have 10 archive urls. 

If at **_first_**  or **_fifth_** iteration it errors out, is this going to stop the loop and return immediately? Is there no need to check until the last url?
^ if that is the expected behaviour, please disregard this comment, :sweat:

Reply:
Ahh nice catch. It should actually be a `continue` and not a `return` since we of course want to continue with the other archives in case we receive an error